### ISSUE TITLE
datasource elasticsearch interval

### ISSUE THREAD

**Issue by coutouly** [grafana/grafana#41685]
**What happened**:
there is a query error in the date_histogram selection 
in date_histogram diagram property  with elasticsearch 8 there is field depreciation that is no longer active
interval
it is replace by
fixed_interval
you can find it in the file:
public/app/plugins/datasource/elasticsearch/datasource.ts

thanks to upgrade to elastic 8.0

 [date_histogram] unknown field [interval] did you mean [fixed_interval]?"
**What you expected to happen**:
query succeed

**How to reproduce it (as minimally and precisely as possible)**:
elasticsearch 8 and date_histogram selection (diagram timeline)

**Anything else we need to know?**:

**Environment**:
- Grafana version: grafana 8.3
- Data source type & version: elasticsearch 8.0 beta
- OS Grafana is installed on: docker
- User OS & Browser: linux chrome
- Grafana plugins: no
- Others:

**Comment by mckn** [comment]
@coutouly thanks for opening this issue. I will forward this to the proper people but please update the issue to follow the issue template.

**Comment by Elfo404** [comment]
closing this as a duplicate of https://github.com/grafana/grafana/issues/19911, thanks for reporting it!

**Comment by joaolgn1985** [comment]
I have the same problem, but when I changed date_histogram to fixed_interval I got another problem `Error Elasticsearch: Missing definition for aggregation [2]`

I updated elasticsearch 7.3 to 8.3.4 and got a lot of major changes

My Grafana is version v7.5.7, maybe it's my problem? I also need to update Grafana too.

**Comment by smalenfant** [comment]
Maybe this will help somebody else. You can add HTTP Headers to tell ES 8.x to accept 7.x queries. It worked for me.

https://www.elastic.co/guide/en/elasticsearch/reference/8.6/rest-api-compatibility.html

**Comment by enewmanMN** [comment]
Can confirm that adding the httpheaders saved the day for me as well in this case :) 

Was seeing both "Elasticsearch error: Bad Gateway" and the error from above about date histogram unknown field interval...

Elastic 8.x
Grafana 8.x
Looking in the grafana data source setup I saw that we were searching with version 7.x defined in the 'elasticsearch details'. 

To fix I FIRST (and unsuccessfully) tried adding application/vnd.elasticsearch+json;compatible-with=8 to accept and content-type custom headers in the grafana data source settings. 
That failed with same interval error. 

Finally I added custom httpheaders accept and content-type set to application/vnd.elasticsearch+json;compatible-with=7 and it worked! 

I think this is related to the fact that we are defining the elasticsearch from grafana as a 7.x search in the data source options -- don't have an option for 8.x there so would need to either upgrade to 9.x to change the search version or add that header to get elastic search to respond with older version :)
![image](https://github.com/grafana/grafana/assets/23745371/ae012288-4b4a-416e-8354-c6827b70f4e7)
